src/core/server/integration_tests/saved_objects/migrations/zdt_v2_compat/create_index.test.ts

```diff
@@ -70,11 +73,11 @@ describe('ZDT with v2 compat - running on a fresh cluster', () => {
     expect(mappingMeta).toEqual({
       docVersions: {
         foo: '10.2.0',
-        legacy: '7.5.0',
+        legacy: '10.0.0',
```

Comment:
This test was not accurate, as the actual `typeRegistry` defaulted to 10.0.0 for almost all the types.
This was done in the `apply_type_defaults.ts` (removed in this PR), in there, all types were set the `switchToModelVersionAt: 8.10.0` if they did not have it.